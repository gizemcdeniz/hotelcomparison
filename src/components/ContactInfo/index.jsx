/* eslint-disable max-len */
import React from "react";
import { Col, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import locationIcon from "../../images/icons/locationIcon.svg";
import phoneIcon from "../../images/icons/phoneIcon.svg";
import mailIcon from "../../images/icons/mailIcon.svg";
import "./index.scss";

export default function ContactInfo() {
  const { t } = useTranslation();

  return (
    <Col className="contactInfo">
      <h2>{t("contactUs.contactUsInfo.header")}</h2>
      <p>{t("contactUs.contactUsInfo.description")}</p>
      <Row className="contactMethods">
        <Col>
          <img className="contactIcon" src={locationIcon} alt="Location icon" />
          <p>{t("contactUs.contactUsInfo.location")}</p>
        </Col>
        <Col>
          <img className="contactIcon" src={phoneIcon} alt="Phone icon" />
          <p>{t("contactUs.contactUsInfo.phone")}</p>
        </Col>
        <Col>
          <img className="contactIcon" src={mailIcon} alt="Contact icon" />
          <p>{t("contactUs.contactUsInfo.email")}</p>
        </Col>
      </Row>
    </Col>
  );
}
