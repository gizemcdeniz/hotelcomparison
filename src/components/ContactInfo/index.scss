@import "../../style/variables";
@import "../../style/headings";
@import "../../style/bootstrapImports";

.contactInfo {
  h2 {
    @include mainHeading;
    margin-bottom: 1rem;
    color: #3d3c3c;
    font-style: italic;
  }

  .contactMethods {
    padding-top: 1rem;
  }

  .contactIcon {
    width: 2rem;
    height: 2rem;
    margin-bottom: 1rem;
  }
}

@include media-breakpoint-down(sm) {
  .contactInfo {
    h2 {
      text-align: center !important;
    }
  }
}
