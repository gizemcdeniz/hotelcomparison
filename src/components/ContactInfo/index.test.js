import React from "react";
import renderer from "react-test-renderer";
import ContactInfo from "./index";

describe("ContactInfo", () => {
  it("renders correctly and matches the snapshot", () => {
    const tree = renderer.create(<ContactInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
