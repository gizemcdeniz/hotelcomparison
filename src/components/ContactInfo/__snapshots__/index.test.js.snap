// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo renders correctly and matches the snapshot 1`] = `
<div
  className="contactInfo col"
>
  <h2>
    contactUs.contactUsInfo.header
  </h2>
  <p>
    contactUs.contactUsInfo.description
  </p>
  <div
    className="contactMethods row"
  >
    <div
      className="col"
    >
      <img
        alt="Location icon"
        className="contactIcon"
        src="locationIcon.svg"
      />
      <p>
        contactUs.contactUsInfo.location
      </p>
    </div>
    <div
      className="col"
    >
      <img
        alt="Phone icon"
        className="contactIcon"
        src="phoneIcon.svg"
      />
      <p>
        contactUs.contactUsInfo.phone
      </p>
    </div>
    <div
      className="col"
    >
      <img
        alt="Contact icon"
        className="contactIcon"
        src="mailIcon.svg"
      />
      <p>
        contactUs.contactUsInfo.email
      </p>
    </div>
  </div>
</div>
`;
